### PR TITLE
add user syncing for kargo

### PR DESCRIPTION
**Pull request by samuelhorwitz**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

Add user syncing to Kargo

### CODE REVIEW

File: modules/kargoBidAdapter.js [hunk 1]
```diff
@@ -53,6 +55,20 @@ export const spec = {
     }
     return bidResponses;
   },
+  getUserSyncs: function(syncOptions) {
+    const syncs = [];
+    const seed = spec._generateRandomUuid();
+    const clientId = spec._getClientId();
+    if (syncOptions.iframeEnabled && seed && clientId) {
+      for (let i = 0; i < SYNC_COUNT; i++) {
```
Comment (jsnellbaker) [inline modules/kargoBidAdapter.js]: Just to confirm the intended logic here - this setup will always generate 5 sync requests that are largely the same minus the `idx` param which gets incremented by 1 for each subsequent sync request.  Is that the expected behavior?

File: test/spec/modules/kargoBidAdapter_spec.js [hunk 2]
```diff
@@ -356,4 +356,81 @@ describe('kargo adapter tests', function () {
...
+  describe('user sync handler', function() {
+    const clientId = '74c81cbb-7d07-46d9-be9b-68ccb291c949';
+    var uid;
+    var shouldSimulateOutdatedBrowser;
+
+    beforeEach(() => {
+      uid = {};
+      shouldSimulateOutdatedBrowser = false;
+
+      sandbox.stub(crypto, 'getRandomValues').callsFake(function(buf) {
```
Comment (jsnellbaker) [inline test/spec/modules/kargoBidAdapter_spec.js]: This currently returns an `undefined` error on the `crypto` object when the unit tests run in IE11 (see the circleci failed job for details).  Can you please take a look and fix the issue?

File: modules/kargoBidAdapter.js (hunk 1 above)
Comment (samuelhorwitz) [reply modules/kargoBidAdapter.js]: Yes, we want to attempt 5 syncs (barring a cap set by the configuration) and we want our server to be able to choose what to sync, potentially randomly, using a stateless system that gets passed an instance ID/seed and an incrementer.

File: test/spec/modules/kargoBidAdapter_spec.js [hunk 3]
```diff
@@ -356,4 +356,99 @@ describe('kargo adapter tests', function () {
       expect(resp).to.deep.equal(expectation);
     });
   });
+
+  describe('user sync handler', function() {
+    const clientId = '74c81cbb-7d07-46d9-be9b-68ccb291c949';
+    var shouldSimulateOutdatedBrowser,
+        uid,
+        isActuallyOutdatedBrowser;
```
Comment (jsnellbaker) [inline test/spec/modules/kargoBidAdapter_spec.js]: There are some extra spaces before these variables that are causing some lint errors when I run `gulp lint`.  See below for a copy of the error message:

```
/Users/jsnellbaker/git/Prebid.js/test/spec/modules/kargoBidAdapter_spec.js
  394:1  error  Expected indentation of 6 spaces but found 8  indent
  395:1  error  Expected indentation of 6 spaces but found 8  indent
```

Can you make this fix?  

Normally linting errors would be caught by circleci.  So going forward, you may want to look into updating/rebasing your fork with the latest version of master so we can catch these automatically when you submit a PR.

File: modules/kargoBidAdapter.js (hunk 1 above)
Comment (jsnellbaker) [reply modules/kargoBidAdapter.js]: Thanks for clarifying.

File: test/spec/modules/kargoBidAdapter_spec.js (hunk 3 above)
Comment (samuelhorwitz) [reply test/spec/modules/kargoBidAdapter_spec.js]: Ah good catch, I'll fix and rebase as well.

**Comment by jsnellbaker** [comment]
@samuelhorwitz Thanks for making the additional changes; LGTM

**Pull request merged**